### ISSUE TITLE
Nativescript 8.3 Angular 13 App crashes when navigating away from listview on IOS platform

### ISSUE THREAD

**Issue by JayLHein** [NativeScript/NativeScript#9919]
### Issue Description

When I navigate away from a listview the app crashes. This only happens on IOS. I have seen similar issues under NS 7 that have been fixed, but not on version 8. I have been searching and working on this problem for a while now. This crash happens fairly consistently.

### Reproduction

Create an app with version 8.x, create a listview, fill with any data, navigate away from the listview page. That's where the crash happens.

### Relevant log output (if applicable)

```shell
====== Assertion failed ======
Native stack trace:
1          0x10d0475dc tns::Assert(bool, v8::Isolate*) + 125
2          0x10cfcd7b8 tns::ArgConverter::Invoke(v8::Local<v8::Context>, objc_class*, v8::Local<v8::Object>, tns::V8Args&, tns::MethodMeta const*, bool) + 102
3          0x10d021c82 tns::MetadataBuilder::InvokeMethod(v8::Local<v8::Context>, tns::MethodMeta const*, v8::Local<v8::Object>, tns::V8Args&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, bool) + 76
4          0x10d02157e tns::MetadataBuilder::MethodCallback(v8::FunctionCallbackInfo<v8::Value> const&) + 222
5          0x10d15b58e v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo) + 606
6          0x10d15aa17 v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, v8::internal::BuiltinArguments) + 727
7          0x10d159f6f v8::internal::Builtin_Impl_HandleApiCall(v8::internal::BuiltinArguments, v8::internal::Isolate*) + 255
8          0x10d918619 Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_BuiltinExit + 57
JavaScript stack trace:
at requestLayout (file: src/webpack:/soberGuide/node_modules/@nativescript/core/ui/core/view/index.ios.js:42:0)
at <anonymous> (file: src/webpack:/soberGuide/node_modules/@nativescript/core/ui/core/properties/index.js:953:0)
at clearInheritedProperties (file: src/webpack:/soberGuide/node_modules/@nativescript/core/ui/core/properties/index.js:1202:0)
at _parentChanged (file: src/webpack:/soberGuide/node_modules/@nativescript/core/ui/core/view-base/index.js:728:36)
at _removeView (file: src/webpack:/soberGuide/node_modules/@nativescript/core/ui/core/view-base/index.js:484:0)
at _removeContainer (file: src/webpack:/soberGuide/node_modules/@nativescript/core/ui/list-view/index.ios.js:412:0)
at ListViewCell.willMoveToSuperview (file: src/webpack:/soberGuide/node_modules/@nativescript/core/ui/list-view/index.ios.js:39:0)
at GC (file: src/webpack:/soberGuide/node_modules/@nativescript/core/utils/index.ios.js:20:0)
at <anonymous> (file: src/webpack:/soberGuide/node_modules/@nativescript/core/utils/utils-common.js:153:52)
at <anonymous> (file: src/webpack:/soberGuide/node_modules/@nativescript/core/utils/utils-common.js:116:0)
```

### Environment

OS: macOS 12.3.1
CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
Shell: /bin/zsh
node: 14.16.0
npm: 6.14.11
nativescript: 8.2.3

# android
java: 1.8.0_242
ndk: Not Found
apis: 28
build_tools: 28.0.3
system_images: Not Found

# ios
xcode: 13.3.1/13E500a
cocoapods: 1.9.0
python: 2.7.18
python3: 3.8.9
ruby: 2.6.8
platforms: 
  - DriverKit 21.4
  - iOS 15.4
  - macOS 12.3
  - tvOS 15.4
  - watchOS 8.5
```

### Dependencies

```json
"dependencies": {
  "@angular/animations": "~13.2.0",
  "@angular/common": "~13.2.0",
  "@angular/compiler": "~13.2.0",
  "@angular/core": "~13.2.0",
  "@angular/forms": "~13.2.0",
  "@angular/platform-browser": "~13.2.0",
  "@angular/platform-browser-dynamic": "~13.2.0",
  "@angular/router": "~13.2.0",
  "@nativescript/angular": "^13.0.0",
  "@nativescript/contacts": "^2.0.4",
  "@nativescript/core": "~8.2.0",
  "@nativescript/geolocation": "^8.0.2",
  "@nativescript/theme": "~2.3.0",
  "nativescript-permissions": "^1.3.12",
  "@nstudio/nativescript-fancyalert": "^4.0.1",
  "date-fns": "^2.19.0",
  "nativescript-phone": "^3.0.3",
  "nativescript-ui-sidedrawer": "~10.0.2",
  "rxjs": "~7.5.0",
  "typescript-string-operations": "^1.4.1",
  "zone.js": "~0.11.5"
},
"devDependencies": {
  "@angular-devkit/build-angular": "~13.2.0",
  "@angular/compiler-cli": "~13.2.0",
  "@nativescript/android": "8.2.2",
  "@nativescript/ios": "8.2.3",
  "@nativescript/types": "~8.2.0",
  "@nativescript/webpack": "~5.0.0",
  "@ngtools/webpack": "~13.2.0",
  "@typescript-eslint/eslint-plugin": "^5.22.0",
  "@typescript-eslint/parser": "^5.22.0",
  "eslint": "^8.14.0",
  "typescript": "~4.5.5"
}
```

### Please accept these terms

- [X] I have searched the [existing issues](https://github.com/NativeScript/NativeScript/issues) as well as [StackOverflow](https://stackoverflow.com/questions/tagged/nativescript) and this has not been posted before
- [X] This is a bug report
- [X] I agree to follow this project's [Code of Conduct](https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#coc)

**Comment by felixkrautschuk** [comment]
I also notice some weird crashes on iOS, mostly when navigating between pages with a RadListView.
However, for me the issue is not related to the core modules of NativeScript (this repo here), but it is related to the [ios v8 runtime](https://github.com/NativeScript/ns-v8ios-runtime/issues/166).

> ====== Assertion failed ======
> Native stack trace:
> 1          0x1134e0adc tns::Assert(bool, v8::Isolate*) + 125
> 2          0x113466cb8 tns::ArgConverter::Invoke(v8::Local<v8::Context>, objc_class*, v8::Local<v8::Object>, tns::V8Args&, tns::MethodMeta const*, bool) + 102
> 3          0x1134bb182 tns::MetadataBuilder::InvokeMethod(v8::Local<v8::Context>, tns::MethodMeta const*, v8::Local<v8::Object>, tns::V8Args&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, bool) + 76
> 4          0x1134baa7e tns::MetadataBuilder::MethodCallback(v8::FunctionCallbackInfo<v8::Value> const&) + 222
> 5          0x1135f488e v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo) + 606
> 6          0x1135f3d17 v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, v8::internal::BuiltinArguments) + 727
> 7          0x1135f326f v8::internal::Builtin_Impl_HandleApiCall(v8::internal::BuiltinArguments, v8::internal::Isolate*) + 255
> 8          0x113db1919 Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_BuiltinExit + 57
> JavaScript stack trace:
> at requestLayout (file: node_modules/@nativescript/core/ui/core/view/index.ios.js:42:0)
> at performLayout (file: node_modules/@nativescript/core/ui/core/view-base/index.js:412:0)
> at requestLayout (file: node_modules/@nativescript/core/ui/core/view-base/index.js:419:0)
> at requestLayout (file: node_modules/@nativescript/core/ui/core/view/view-common.js:735:0)
> at requestLayout (file: node_modules/@nativescript/core/ui/core/view/index.ios.js:38:0)
> at performLayout (file: node_modules/@nativescript/core/ui/core/view-base/index.js:412:0)
> at requestLayout (file: node_modules/@nativescript/core/ui/core/view-base/index.js:419:0)
> at requestLayout (file: node_modules/@nativescript/core/ui/core/view/view-common.js:735:0)
> at requestLayout (file: node_modules/@nativescript/core/ui/core/view/index.ios.js:38:0)
> at performLayout (file: node_modules/@nativescript/core/ui/core/view-base/index.js:412:0)

Unfortunately I am not able to create an app for reproduction. 

I just noticed that the issue does not occur anymore if I remove some parts from the UI, for example the RadListView-header (which has some complex layout in some cases but no way to simplify anymore), the Image inside listitem-template, ... but I have no clue why I need to remove UI components to make the app not crashing anymore.

Please note: **When switching back to @nativescript/ios 8.0.0, the issue does NOT occur!**
However, I need to switch back to xcode 13.2 to use that version of the ios runtime, which is pretty annoying.

**Comment by felixkrautschuk** [comment]
Just to add one thing:
In my case, commenting the call of [queueGC](https://github.com/NativeScript/NativeScript/blob/master/packages/core/ui/image/index.ios.ts#L38) inside disposeImageSource method of the core/ui/image module drastically decreases the amount of crashes while testing.

Not sure why this line was not causing any crashes when using @nativescript/ios 8.0.0 though.

**Comment by JayLHein** [comment]
I followed the lead from felixkrautschuk and using @nativescript/ios 8.0.0 and xcode 13.2. With this combination there are no crashes. So something changed between Nativescript/ios 8.0 and 8.3 that is causing the crashes. Thanks to felixkrautschuk for pointing this out.

**Comment by 3rror404** [comment]
Can also confirm downgrading to @nativescript/ios 8.0.0 eliminates the crashes I have been fighting against.

**Comment by felixkrautschuk** [comment]
@JayLHein @3rror404 thanks for confirming that you also have no problems using @nativescript/ios 8.0.0.

For me the last version without crashing was **8.0.1-rc.0** and the first version crashing is 8.1.0-alpha.0. Maybe you can check if you can confirm that as well? Just to narrow down the cause of the problem a little further.

**Comment by felixkrautschuk** [comment]
I finally managed to create a sample app for reproducing this issue. In our production app there are many different scenarios where the app is crashing (starting from @nativescript/ios 8.1.x) and at least I could isolate one scenario in that sample app here.

[ns-ios-v8-crash.zip](https://github.com/NativeScript/NativeScript/files/8926090/ns-ios-v8-crash.zip)

Steps to reproduce:
- start app
- navigate to the RadListView page
- tap any item
- go back
- tap any item
- go back
- tap any item -> crash

https://user-images.githubusercontent.com/6443021/174264466-a7cf0daf-9918-4f0e-8cd9-717373a3a0e2.mov

------------------------

Stacktrace:

> ====== Assertion failed ======
> Native stack trace:
> 1          0x10c14c5dc tns::Assert(bool, v8::Isolate*) + 125
> 2          0x10c0d27b8 tns::ArgConverter::Invoke(v8::Local<v8::Context>, objc_class*, v8::Local<v8::Object>, tns::V8Args&, tns::MethodMeta const*, bool) + 102
> 3          0x10c126c82 tns::MetadataBuilder::InvokeMethod(v8::Local<v8::Context>, tns::MethodMeta const*, v8::Local<v8::Object>, tns::V8Args&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, bool) + 76
> 4          0x10c12657e tns::MetadataBuilder::MethodCallback(v8::FunctionCallbackInfo<v8::Value> const&) + 222
> 5          0x10c26058e v8::internal::FunctionCallbackArguments::Call(v8::internal::CallHandlerInfo) + 606
> 6          0x10c25fa17 v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<false>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, v8::internal::BuiltinArguments) + 727
> 7          0x10c25ef6f v8::internal::Builtin_Impl_HandleApiCall(v8::internal::BuiltinArguments, v8::internal::Isolate*) + 255
> 8          0x10ca1d619 Builtins_CEntry_Return1_DontSaveFPRegs_ArgvOnStack_BuiltinExit + 57
> 9          0x10c9b638a Builtins_InterpreterEntryTrampoline + 202
> JavaScript stack trace:
> at requestLayout (file: app/webpack:/ns-ios-v8-crash/node_modules/@nativescript/core/ui/core/view/index.ios.js:42:0)
> at performLayout (file: app/webpack:/ns-ios-v8-crash/node_modules/@nativescript/core/ui/core/view-base/index.js:412:0)
> at requestLayout (file: app/webpack:/ns-ios-v8-crash/node_modules/@nativescript/core/ui/core/view-base/index.js:419:0)
> at requestLayout (file: app/webpack:/ns-ios-v8-crash/node_modules/@nativescript/core/ui/core/view/view-common.js:735:0)
> at requestLayout (file: app/webpack:/ns-ios-v8-crash/node_modules/@nativescript/core/ui/core/view/index.ios.js:38:0)
> at performLayout (file: app/webpack:/ns-ios-v8-crash/node_modules/@nativescript/core/ui/core/view-base/index.js:412:0)
> at requestLayout (file: app/webpack:/ns-ios-v8-crash/node_modules/@nativescript/core/ui/core/view-base/index.js:419:0)
> at requestLayout (file: app/webpack:/ns-ios-v8-crash/node_modules/@nativescript/core/ui/core/view/view-common.js:735:0)
> at requestLayout (file: app/webpack:/ns-ios-v8-crash/node_modules/@nativescript/core/ui/core/view/index.ios.js:38:0)
> at setLocalValue (file: app/webpack:/ns-ios-v8-crash/node_modules/@nativescript/core/ui/core/properties/index.js:551:0)

------------------------

My config:
- NativeScript CLI 8.2.3
- @nativescript/core 8.2.5
- nativescript-ui-listview 10.1.1
- @nativescript/ios 8.3.0 (crash reproducable since 8.1.0)
- @nativescript/webpack 5.0.6

------------------------

Some further observations:
- when using @nativescript/ios 8.1.0, you have to navigate back and forth a few times more often to reproduce the crash
- using @nativescript/ios 8.0.0, no crashes are happening
- when switching from RadListView to ListView, the provided sample app also does not crash anymore, HOWEVER there are some crashes in our real app without RadListView involved, and also @JayLHein noticed crashes using the default ListView
- the weirdest thing ever: the "detail"-page has the following UI
```
...
<StackLayout padding="20">
        <Label text="TEST" tap="myTestFunc"/>
        <Label text="TEST 2" visibility="{{ isLoading ? 'collapse' : 'visible' }}"/>
</StackLayout>
...
```
When removing the `tap="myTestFunc"` from the first Label OR when removing the visibility statement from the second Label, the crashes also do not occur anymore .... ???

So it is actually impossible to adjust our app to make it work using the latest iOS v8 runtime (to be able using latest xcode, test on latest iOS, ...).
No idea what changes from the [8.1.0 release](https://github.com/NativeScript/ns-v8ios-runtime/releases/tag/v8.1.0) could cause this.

@NathanaelA @rigor789 would be nice if you (or anymone else involved in the iOS runtime) could take a look.

**Comment by pichurichko** [comment]
+1 vote
This issue is a stopper for upgrade to latest Xcode and iOS NS platform version
My logs and cases are similar to provided above

**Comment by prabudevarrajan** [comment]
I am also facing the same problem. Same stackstrace mentioned by @felixkrautschuk. Stopper bug preventing upgrade to lasted ios/NS platform.

**Comment by tweb-ix** [comment]
I get the same stacktrace as JayLHein, with the difference that our app also crashes with @nativescript/ios 8.0.0

**Comment by felixkrautschuk** [comment]
Just an update: I need to test further first, but it looks like upgrading to @nativescript/core 8.3.0 fixes this issue. At least I was not able to reproduce the crashes anymore in the sample app that I provided above.
I also was not able to see those crashes in our real app, but as I said, I need to continue testing. Just wanted to give a quick update.

**Comment by CatchABus** [comment]
@felixkrautschuk I was also facing issues because of the exact same `queueGC` calls and it seems that proper iOS native view disposal helped on solving them for me.
This is the patch included in 8.3: https://github.com/NativeScript/NativeScript/commit/f548fdc73512bb74e7ec500550e2db0166e7237b

Perhaps, you can experiment with this and see if it actually fixes your errors too.

**Comment by NathanWalker** [comment]
Confirmed resolved in 8.3
